File: slot-dart/lib/src/slot_engine.dart
library slot.slot_engine;

import 'core/engine.dart';
import 'slot_game.dart';

class SlotEngine extends Engine {

  static void init(SlotGame game) {

    print('SlotEngine.init');

    Engine.init(game);

  }

  static SlotGame get game {
    return _game as SlotGame;
  }

}

File: slot-ts/src/SlotEngine.ts
import Engine from './engine/core/Engine';
import SlotGame from './SlotGame';
import Stage from 'openfl/display/Stage';

export default class SlotEngine extends Engine {
  public constructor() {
    super(SlotGame);
  }
}

function init() {
  const engine = new SlotEngine();
  document.body.appendChild(engine.stage.element);
}

function createStartButton() {
  const button = document.createElement('button');
  button.textContent = 'Start';
  document.body.appendChild(button);
  button.addEventListener('click', () => {
    document.body.removeChild(button);
    init();
  });
}

init();
